src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionId.java

```diff
@@ -17,15 +17,28 @@
 
 import com.google.auto.value.AutoValue;
 import com.google.devtools.build.lib.cmdline.Label;
+import java.util.Optional;
 
 /** A unique identifier for a {@link ModuleExtension}. */
 @AutoValue
 public abstract class ModuleExtensionId {
+
+  static final class Namespace {
+    final ModuleKey module;
+
+    Namespace(ModuleKey module) {
+      this.module = module;
+    }
+  }
+
   public abstract Label getBzlFileLabel();
 
   public abstract String getExtensionName();
 
-  public static ModuleExtensionId create(Label bzlFileLabel, String extensionName) {
-    return new AutoValue_ModuleExtensionId(bzlFileLabel, extensionName);
+  public abstract Optional<Namespace> getNamespace();
```

Comment:
IMO we don't need to generalize the concept of a "module extension ID namespace" until that's actually happened. For now, just calling this `isolatedToModuleKey` or some such is probably clearer.